Backend/orders/ordersController.js

```diff
@@ -0,0 +1,87 @@
+const ordersModel = require("./ordersModel.js");
+const chatModel = require("../chat/chatModel.js");
+
+const { mapOrderBody } = require("./ordersHelper");
+
+exports.createOrder = async (req, res) => {
+  const { order, initial_message } = req.body;
+  try {
+    const savedOrder = await ordersModel.saveOrder(order);
+    if (!savedOrder) {
+      return res.status(400).json({
+        errorMessage: "Something went wrong with your trade request",
+      });
+    }
+
+    const messageBody = {
+      order_id: savedOrder.id,
+      text: initial_message,
+      author_id: savedOrder.taker_id,
+    };
+
+    const newMessage = await chatModel.saveMessage(messageBody);
```

Comment:
I think you can insert everything in one query. There are two ways to do that

1) Wrap it up in with `WITH`
2) Use a psql transaction and make a few statements

https://stackoverflow.com/questions/20561254/insert-data-in-3-tables-at-a-time-using-postgres

Reply:
Yes, we can do that, but there are several issues that will occur if we do this

1: The two query functions are in two separate files.
2: If we combine it in different files, then the code will be duplicated.
3. What if we use an individual query? For example, if you just want to call the saveMessage query, in this case both queries run and others are re-run unnecessarily.
4: disturb the structure of code and create complexity when you mix up two different entity (tables) queries in the same place.

Reply:
ok

Reply:
Check